cmd/showbudget/show_budget_test.go

```diff
@@ -19,16 +21,19 @@ var _ = gc.Suite(&showBudgetSuite{})
 
 type showBudgetSuite struct {
 	coretesting.FakeJujuXDGDataHomeSuite
-	stub    *testing.Stub
-	mockAPI *mockapi
+	stub          *testing.Stub
+	mockBudgetAPI *mockBudgetAPI
+	mockAPI       *mockAPI
 }
 
 func (s *showBudgetSuite) SetUpTest(c *gc.C) {
 	s.CleanupSuite.SetUpTest(c)
 	s.FakeJujuXDGDataHomeSuite.SetUpTest(c)
 	s.stub = &testing.Stub{}
-	s.mockAPI = &mockapi{s.stub}
-	s.PatchValue(showbudget.NewAPIClient, showbudget.APIClientFnc(s.mockAPI))
+	s.mockBudgetAPI = &mockBudgetAPI{s.stub}
+	s.mockAPI = &mockAPI{s.stub}
+	s.PatchValue(showbudget.NewBudgetAPIClient, showbudget.BudgetAPIClientFnc(s.mockBudgetAPI))
+	s.PatchValue(showbudget.NewAPIClient, showbudget.NewAPIClientFnc(s.mockAPI))
 }
 
 func (s *showBudgetSuite) TestShowBudgetCommand(c *gc.C) {
```

Comment:
Add a test where resolving model names returns an error.
Also a test where some models have no info would be good.